airflow/jobs/scheduler_job.py

```diff
@@ -145,6 +145,12 @@ def __init__(
 
         self.dagbag = DagBag(dag_folder=self.subdir, read_dags_from_db=True, load_op_links=False)
 
+        if conf.getboolean('smart_sensor', 'use_smart_sensor'):
+            warnings.warn(
+                'Smart sensors are deprecated. Please use Deferrable Operators instead.',
```

Comment:
Do you'll think adding a link to the docsite for deferrable operators will be helpful?

@potiuk @turbaszek @eladkal ?

Reply:
Yep . As described in the devlist. We definitely need a place in the doc (and linked from here) where we explain (shortly):

1) why we cannot provide precise migration instructions
2) what should be general approach of the user who has to migrate. Something alongside "For every smart sensor you use, you have to either use or write your own defferable operator if want to achieve similar results. If you want to know how to write your own operator - find it [here]" (another link)

We should simply assume that the users have not read the docs, know nothing about deferrable operators and they should understand that they need to write they own deferrable operator instead BEFORE they read the whole documentation (which we shoudl helpfuly link so that they do not have to google it or search the documentation).

That's teh "approach I 'd use for any deprecation" . I think those two pieces are crucial:

1) Explaining "why" we cannot give them instructions
2) Explaining 'what" they need to do before actually reading the documentation is crucial.

The 1) prevents questions "Please give me exact instructions on how to remove the deprecation?"
The 2) prevents questions "What do I need to to do now?".

That's it. No more, no less. 

It's simply giving the right answers to the anticipated questions straight there when you see the problem.

Reply:
>It's simply giving the right answers to the anticipated questions straight there when you see the problem.

agree with a single word replacement.

"right" -> "all"

"right" is relative please

Reply:
If we can be more explicit about what needs to be done - we should.
However I believe that if we know that a feature is deprecated we should say it as soon as possible - regardless if we have all the needed docs / explanations. At the end the explanations are for users who need to migrate from the deprecated features to the new one BUT there are also users who may just now playing with the feature and thinking to start use it for them - we just need to say that it's deprecated. They don't need migration docs.

So this is a tradeoff. i'm happy with what we have so far. We can always add more information.

Also want to add that we've been in this case before :) 
TaskGroups <-> SubDags are also not 1-to-1